accepted/future-releases/enhanced-enums/feature-specification.md

```diff
@@ -54,12 +54,16 @@ It is a **compile-time error** if the initializer list of a non-redirecting gene
 
 _We will introduce the necessary super-invocation ourselves as an implementation detail. From the user’s perspective, they extend `Enum` which has no public constructors. We could allow `super()`, which would then be a constructor of `Enum`, but it's simpler to just disallow super invocations entirely._
 
+It is a **compile-time error** if the argument list of a non-redirecting generative constructor includes a `super` parameter.
+
+_Super parameters (a feature introduced in the same release as enhanced enums) implicitly adds arguments to the `super` constructor invocation of a non-redirecting generative constructor, and such enum constructors do not have a known superclass constructor expecting such arguments._
```

Comment:
If line 57 reports that an error is known to occur (based on existing rules) then it should presumably be commentary.

Reply:
We can have this as a direct rule on super-parameters, rather than here, but I'd mention it here as a rule as well.
(In super parameters, we say that it's a compile-time error to have a super parameter in any constructor which is not a non-redirecting generative constructor, and we should add "of a `class` declaration" now that we can have constructors in `enum` declarations too.)

This is not *just* an error because the desugaring would have an error. It's that too, but not just.

Reply:
Right, that makes sense, we do want to have errors that are specified directly rather than "desugar this, and then backtrack any errors to the enum".